core/primitives/src/challenge.rs

```diff
@@ -12,6 +12,10 @@ pub type StateItem = Vec<u8>;
 
 pub type PartialState = Vec<StateItem>;
 
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, Eq, PartialEq)]
+// TODO MOO this is weird
```

Comment:
Might as well define

`pub struct PartialState(Vec<StateItem>)`

and only keep one?

Reply:
++